MISC: Remove a carriage return char at each line end if it exists

mmatsuya opened the pull request:
Powershell returns \r\n for a new line, and \r can be left after processing the ssh result as below. It causes ValueError in attrs_parse.

lines = ['\r', '\r', 'DistinguishedName : CN=Domain Admins,CN=Users,DC=ad-n97b,DC=test\r',...
line       = '\r'

```
>           (key, value) = map(lambda x: x.lstrip(), line.split(":", 1))
E           ValueError: not enough values to unpack (expected 2, got 1)
```
This removes the carriage return char in attrs_parse before checking attr.

mmatsuya commented:
This is related to Nightly-Z test for RHEL9.4. (2024-06-18_02-07/tier-1/pytest-mh-alltests-high/6/runner.log).

pbrezina commented:
Thank you. Can this be handled in attrs_parse directly?

mmatsuya commented:
Yes, I can. I changed the commit to handle it in attrs_parse.

jakub-vavra-cz reviewed (approved):
Thanks for the fix.

jakub-vavra-cz commented:
@pbrezina: Please merge this as You are the only one that can do that due to the red ci.

mmatsuya commented:
I changed it again to remove \r only with:
>  line = lines[i].rstrip('\r').

Because it failed in test_attrs_parse__long_line of test_misc.py. rstrip() with removing the last space of "description: My teacher is a good ". The test should work with this change.

pbrezina commented:
Please, run `black .` locally on the code to fix linter error.

pbrezina commented:
> @pbrezina: Please merge this as You are the only one that can do that due to the red ci.

Red CI indicates a problem, please, start clicking on the Details link prior merging things...

jakub-vavra-cz commented:
@pbrezina Well id did click it and it is in tox:docs which seems completely unrelated and the same failures are on other PRs as well..

mmatsuya commented:
> @mmatsuya Please, run black . locally on the code to fix linter error. It would be also nice if you can extend the unit tests for this function.

@pbrezina. I fixed the linter error, and I added another test case for test_attrs_parse__long_line test. Let me know if there is anything else I should do. Thanks!

danlavu commented:
Tested, this fixes the gpo test failures, happy to approve after tox is green.

pbrezina commented:
There was also `./tests/test_misc.py:76:14: E201 whitespace after '['`. I fixed that and we should be good to go.

pbrezina reviewed (approved):
Thanks guys for working on this.

jakub-vavra-cz reviewed (commented):
LGTM
